### PR TITLE
Support Network APIs

### PR DESCRIPTION
**Pull request by akeely**
Hi,

I need support for the network APIs (#356) for a project that I'm working on. I'm happy to contribute the code that I write, but I can't get the integration tests to run locally (I assume they are hanging at some point - I left them running overnight and they did not complete).

This pull request only contains one of the APIs (ListNetworks). If this is useful, I can continue to add the remaining APIs.

Let me know if I should continue. Thanks!
Andrew

### CODE REVIEW

File: src/main/java/com/github/dockerjava/api/model/Network.java [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+        private String endpointId;
+
+        @JsonProperty("MacAddress")
+        private String macAddress;
+
+        @JsonProperty("IPv4Address")
+        private String ipV4Address;
+
+        @JsonProperty("IPv6Address")
+        private String ipV6Address;
```
Comment (KostyaSha) [inline src/main/java/com/github/dockerjava/api/model/Network.java]: Maybe `v` should be lower case?

File: src/main/java/com/github/dockerjava/jaxrs/DockerCmdExecFactoryImpl.java [hunk 2]
```diff
@@ -16,6 +16,7 @@
 import javax.ws.rs.client.ClientResponseFilter;
 import javax.ws.rs.client.WebTarget;
 
+import com.github.dockerjava.api.command.*;
```
Comment (KostyaSha) [inline src/main/java/com/github/dockerjava/jaxrs/DockerCmdExecFactoryImpl.java]: Don't replace single imports with hash imports.

File: src/main/java/com/github/dockerjava/netty/DockerCmdExecFactoryImpl.java [hunk 3]
```diff
@@ -1,6 +1,9 @@
 package com.github.dockerjava.netty;
 
 import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.github.dockerjava.api.command.*;
+import com.github.dockerjava.netty.exec.*;
```
Comment (KostyaSha) [inline src/main/java/com/github/dockerjava/netty/DockerCmdExecFactoryImpl.java]: the same, don't replace single imports with hash imports.

File: src/test/java/com/github/dockerjava/core/TestDockerCmdExecFactory.java [hunk 4]
```diff
@@ -6,52 +6,8 @@
 import java.util.List;
 
 import com.github.dockerjava.api.async.ResultCallback;
-import com.github.dockerjava.api.command.AttachContainerCmd;
+import com.github.dockerjava.api.command.*;
```
Comment (KostyaSha) [inline src/test/java/com/github/dockerjava/core/TestDockerCmdExecFactory.java]: .

**Comment by KostyaSha** [comment]
Some tests would be extremely critical

**Comment by marcuslinke** [comment]
@akeely Thanks for your contribution. I want to include the network API in the upcoming v3.0.0 release,  so I've created a branch https://github.com/docker-java/docker-java/tree/issue-356 to work on it. All network API related PRs should be targeted against it. I've added a simple test for the list networks functionaltity. Would be great if you could implement at least some of the missing APIs. If you have trouble to get the integration tests running I could add the missing tests.

**Comment by marcuslinke** [comment]
Merged into https://github.com/docker-java/docker-java/tree/issue-356 for later merge into master. Closing this now.

**Pull request closed without merging**